haystack/nodes/file_converter/markdown.py

```diff
@@ -40,32 +72,53 @@ def convert(
             attributes. If you want to ensure you don't have duplicate documents in your DocumentStore but texts are
             not unique, you can modify the metadata and pass e.g. `"meta"` to this field (e.g. [`"content"`, `"meta"`]).
             In this case the id will be generated by using the content and the defined metadata.
+        :param remove_code_snippets: Whether to remove snippets from the markdown file.
+        :param extract_headlines: Whether to extract headings from the markdown file.
         """
-        if id_hash_keys is None:
-            id_hash_keys = self.id_hash_keys
+
+        id_hash_keys = id_hash_keys if id_hash_keys is not None else self.id_hash_keys
+        remove_code_snippets = remove_code_snippets if remove_code_snippets is not None else self.remove_code_snippets
```

Comment:
@bogdankostic another option is `id_hash_keys = id_hash_keys or self.id_hash_keys`, but I am undecided about which one is better.

Reply:
`id_hash_keys = id_hash_keys or self.id_hash_keys` would work in this case because `id_hash_keys`  is supposed to be a list, but this doesn't work with optional boolean parameters.

For example, if we explicitly set `remove_code_snippets` to `False` when calling `convert`, `remove_code_snippets or self.remove_code_snippets` would evaluate to whatever value `self.remove_code_snippets` has.

For consistency, I would like all these lines to have the same pattern.

Reply:
Great points